src/t_zset.c

```diff
@@ -2719,6 +2719,10 @@ void zunionInterDiffGenericCommand(client *c, robj *dstkey, int numkeysIndex, in
 
                     /* We stop the searching after reaching the limit. */
                     if (limit && cardinality >= limit) {
+                        /* Cleanup before we break the zuiNext loop. */
+                        if (zval.flags & OPVAL_DIRTY_SDS)
+                            sdsfree(zval.ele);
```

Comment:
maybe we better wrap this logic in a function, and also call it from `zuiNext`
how about zuiDiscardDirtyValue?

Reply:
done